src/store/activities/actions.js

```diff
@@ -1,17 +1,28 @@
-import uuid from 'uuid-v4';
+import database from '../../firebase/firebase';
 
-export const addActivity = ({ classNo, day, name, room, teacher } = {}) => ({
+export const addActivity = activity => ({
   type: 'ADD_ACTIVITY',
-  activity: {
-    classNo,
-    day,
-    id: uuid(),
-    name,
-    room,
-    teacher,
-  },
+  activity,
 });
 
+export const startAddActivity = (activityData = {}) => {
+  return dispatch => {
+    const { classNo, day, name, room, teacher } = activityData;
+    const activity = { classNo, day, name, room, teacher };
+    return database
+      .ref('activities/items')
+      .push(activity)
+      .then(ref => {
```

Comment:
you don't have `catch` if error occurres. 

Try to do in `async/awat` fashion, using try-catch block. You can do it in separate PR